--- a/drivers/crypto/Kconfig
+++ b/drivers/crypto/Kconfig
@@ -797,4 +797,6 @@ config CRYPTO_DEV_SA2UL
 source "drivers/crypto/aspeed/Kconfig"
 source "drivers/crypto/starfive/Kconfig"
 
+source "drivers/crypto/siflower/Kconfig"
+
 endif # CRYPTO_HW
--- a/drivers/crypto/Makefile
+++ b/drivers/crypto/Makefile
@@ -39,6 +39,7 @@ obj-$(CONFIG_CRYPTO_DEV_S5P) += s5p-sss.
 obj-$(CONFIG_CRYPTO_DEV_SA2UL) += sa2ul.o
 obj-$(CONFIG_CRYPTO_DEV_SAHARA) += sahara.o
 obj-$(CONFIG_CRYPTO_DEV_SL3516) += gemini/
+obj-$(CONFIG_CRYPTO_DEV_SIFLOWER) += siflower/
 obj-y += stm32/
 obj-$(CONFIG_CRYPTO_DEV_TALITOS) += talitos.o
 obj-$(CONFIG_CRYPTO_DEV_VIRTIO) += virtio/
